clients/linux/src/app.rs

```diff
@@ -926,6 +931,154 @@ impl LbApp {
+            *progress.borrow_mut() += 1;
+            pbar.set_fraction(*progress.borrow() as f64 / total as f64);
+
+            match maybe_info {
+                None => {
+                    d.close();
+                    m.send(Msg::RefreshTree);
+                }
+                Some(info) => {
+                    lb_lbl.set_text(&info.lockbook_path);
+                    disk_lbl.set_text(&format!("{}", info.disk_path.display()));
+                    prog_lbl.set_text(&format!("{}/{}", *progress.borrow(), total));
+                }
+            }
+            glib::Continue(true)
+        });
+
+        let f = closure!(ch => move |progress: ImportExportFileInfo| {
+            ch.send(Some(progress)).unwrap();
+        });
+
+        spawn!(self.core as c, self.messenger as m => move || {
+            for uri in uris {
+                if let Some(path) = uri.strip_prefix(FILE_SCHEME) {
+                    if let Err(err) = c.import_file(parent, path.to_string(), Some(Box::new(f.clone()))) {
+                        m.send_err_dialog("Import files", err);
+                        break;
+                    };
+                } else {
+                    eprintln!("A uri scheme besides {} got through somehow!", FILE_SCHEME);
+                    return;
+                }
+            }
+
+            ch.send(None).unwrap();
+        });
+
+        Ok(())
+    }
+
+    fn show_dialog_export_file(&self) -> LbResult<()> {
+        let (selected_tpaths, model) = self.gui.account.sidebar.tree.selected_rows();
+
+        let iters = selected_tpaths
+            .iter()
+            .map(|selected| model.get_iter(selected).unwrap())
+            .collect::<Vec<GtkTreeIter>>();
+        let ids = iters
+            .iter()
+            .map(|iter| Uuid::parse_str(&tree_iter_value!(model, iter, 2, String)).unwrap())
+            .collect::<Vec<Uuid>>();
+
+        let d = FileChooserDialog::new(None, Some(&self.gui.win), FileChooserAction::SelectFolder);
+
+        d.add_buttons(&[
+            ("Cancel", GtkResponseType::Cancel),
+            ("Select", GtkResponseType::Ok),
+        ]);
+
+        let resp = d.run();
+        d.close();
+
+        if resp == GtkResponseType::Ok {
+            let dest = d
+                .get_filename()
+                .unwrap()
+                .into_os_string()
+                .into_string()
+                .map_err(LbError::fmt_program_err)?;
+
+            let (load_d, lb_lbl, disk_lbl, prog_lbl, pbar) =
+                self.gui.new_import_export_dialog(false);
+
+            load_d.show_all();
+
+            let mut total = 0;
+            let progress = Rc::new(RefCell::new(-1));
+
+            for id in ids.clone() {
+                let meta = self.core.file_by_id(id)?;
+
+                total += match meta.file_type {
+                    FileType::Document => 1,
+                    FileType::Folder => match self.core.get_children_recursively(id) {
```

Comment:
```suggestion
            for id in &ids {
                let meta = self.core.file_by_id(*id)?;

                total += match meta.file_type {
                    FileType::Document => 1,
                    FileType::Folder => match self.core.get_children_recursively(*id) {
```
Correct me if I'm wrong, but I don't see a good reason to clone here.